custom_typings/gb-tracker-client.d.ts

```diff
@@ -1,19 +1,91 @@
 declare module 'gb-tracker-client' {
-  interface TrackerClient {
+  class TrackerClient {
+    constructor(customerId: string, area: string);
+
     enableWarnings(): void;
     disableWarnings(): void;
-    setStrictMode(): void;
+    setStrictMode(strict: boolean): void;
 
-    sendAutoSearchEvent(data: any): void;
-    sendAddToCartEvent(data: any): void;
-    sendOrderEvent(data: any): void;
-    sendSearchEvent(data: any): void;
-    sendViewProductEvent(data: any): void;
+    autoSetVisitor(visitorId?: string): void;
     setVisitor(visitorId: any, sessionId: any): void;
-    new (customerId: any, area: any);
+    getVisitorId(): string;
+    getSessionId(): string;
+    getLoginId(): string;
+
+    setInvalidEventCallback(cb: () => void): void;
+
+    sendAddToCartEvent(data: TrackerClient.CartEvent): void;
+    sendViewCartEvent(data: TrackerClient.CartEvent): void;
+    sendRemoveFromCartEvent(data: TrackerClient.CartEvent): void;
```

Comment:
in the typings for the tracker client, so these are just to tell us about which methods it exposes